addons/mail/static/src/discuss/call/common/call_menu.js

```diff
@@ -9,6 +9,38 @@ export class CallMenu extends Component {
     setup() {
         super.setup();
         this.rtc = useState(useService("discuss.rtc"));
+        this.state = useState({
+            systrayIcon: undefined,
+        });
+        this.lastActions = [];
+        this.actionsIcons = [
+            "fa-desktop",
+            "fa-video-camera",
+            "fa-hand-paper-o",
+            "fa-deaf",
+            "fa-microphone-slash",
+        ];
+        useEffect(
+            (...actions) => {
+                for (let i = 0; i < actions.length; i++) {
+                    if (actions[i] !== this.lastActions[i]) {
+                        this.state.systrayIcon = actions[i]
+                            ? this.actionsIcons[i]
+                            : this.actionsIcons[actions.findIndex((v) => v)];
+                        break;
```

Comment:
So if given state becomes falsy, it picks the 1st action being truthy:
- enable screen sharing
- activate camera
- mute microphone
- unmute microphone

=> would expect to see camera icon. But the 1st truthy action in list is screen sharing.


To make sure the last action is picked, better model active actions as an array where items are `.unshift()`, so that `array[0]` always represent the last active action.